include/range/v3/algorithm/for_each_n.hpp

```diff
@@ -0,0 +1,67 @@
+/// \file
+// Range v3 library
+//
+//  Copyright Eric Niebler 2014
+//  Copyright Rostislav Khlebnikov 2017
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+//
+#ifndef RANGES_V3_ALGORITHM_FOR_EACH_N_HPP
+#define RANGES_V3_ALGORITHM_FOR_EACH_N_HPP
+
+#include <functional>
+#include <range/v3/range_fwd.hpp>
+#include <range/v3/range_traits.hpp>
+#include <range/v3/range_concepts.hpp>
+#include <range/v3/algorithm/tagspec.hpp>
+#include <range/v3/utility/functional.hpp>
+#include <range/v3/utility/iterator_traits.hpp>
+#include <range/v3/utility/static_const.hpp>
+#include <range/v3/utility/tagged_pair.hpp>
+
+namespace ranges
+{
+    inline namespace v3
+    {
+        /// \addtogroup group-algorithms
+        /// @{
+        struct for_each_n_fn
+        {
+            template<typename I, typename F, typename P = ident,
+                CONCEPT_REQUIRES_(InputIterator<I>() &&
+                    IndirectInvocable<F, projected<I, P>>())>
+            tagged_pair<tag::in(I), tag::fun(F)>
```

Comment:
(In the meanwhile I'll continue reading the range proposal - maybe the answer is there :) )